log/nsq_aggregator.go

```diff
@@ -0,0 +1,105 @@
+package log
+
+import (
+	"fmt"
+	"time"
+
+	nsq "github.com/nsqio/go-nsq"
+
+	"github.com/deis/logger/osutils"
+	"github.com/deis/logger/storage"
+)
+
+type nsqAggregator struct {
+	listening      bool
+	consumer       *nsq.Consumer
+	messageHandler *nsqMessageHandler
+}
+
+func newNsqAggregator(storageAdapter storage.Adapter) (Aggregator, error) {
+	return &nsqAggregator{
+		messageHandler: newNsqMessageHandler(storageAdapter),
+	}, nil
+}
+
+// Listen starts the aggregator
+func (a *nsqAggregator) Listen() error {
+	// Should only ever be called once
+	if !a.listening {
+		a.listening = true
+		host := osutils.GetStringOpt("DEIS_NSQD_SERVICE_HOST", "")
+		port, err := osutils.GetIntOpt("DEIS_NSQD_SERVICE_PORT_TRANSPORT", 4150)
+		if err != nil {
+			return nil
+		}
+		url := fmt.Sprintf("%s:%s", host, port)
```

Comment:
Yep. I will fix. Port used to be a string and that line would have been correct... this was a careless copy/paste error.